cppForSwig/Makefile.am

```diff
@@ -0,0 +1,110 @@
+LINK = $(CXX)
+
+PYTHON_INCLUDES=@PYTHON_INCLUDES@
+PYTHON_LIBS=@PYTHON_LIBS@
+
+if TARGET_LINUX
+UNAME=Linux
+else
+if TARGET_WINDOWS
+UNAME=Windows
+else
+if TARGET_DARWIN
+UNAME=Darwin
+endif
+endif
+endif
+
+# Export variables to make them available to cryptopp Makefile.
+export AR
+export CXX
+export RANLIB
+export UNAME
+
+CRYPTOPP_EXT=.a
+
+if TARGET_DARWIN
+MACOSX_DEPLOYMENT_TARGET=10.7
+export MACOSX_DEPLOYMENT_TARGET
+LDFLAGS += -undefined dynamic_lookup -headerpad_max_install_names
+endif
+
+OBJS = UniversalTimer.o BinaryData.o lmdb_wrapper.o StoredBlockObj.o \
+	BtcUtils.o BlockObj.o BlockUtils.o EncryptionUtils.o \
+	BtcWallet.o LedgerEntry.o ScrAddrObj.o Blockchain.o BlockWriteBatcher.o \
+	BDM_mainthread.o lmdbpp.o BDM_supportClasses.o \
+	BlockDataViewer.o HistoryPager.o Progress.o \
+	libcryptopp$(CRYPTOPP_EXT) mdb.o midl.o txio.o FileMap.o SSHheaders.o
+
+CPPFLAGS = $(ARMORY_CPPFLAGS) -Icryptopp -Imdb -DUSE_CRYPTOPP -D__STDC_LIMIT_MACROS
+LDLIBS = -lpthread -Lmdb
+SWIG_OPTS    = -c++ -python -threads
+
+SWIG_INC     =
+
+CXXCPP = $(CPPFLAGS) $(PYTHON_INCLUDES)
+
+if TARGET_DARWIN
+CXXCPP += -stdlib=libc++
+LDLIBS += -lc++
+if BUILD_LINUX
+CXXCPP += -I/usr/include/c++/v1
```

Comment:
I installed clang from the llvm website and it seems to work now without any explicit listing of include paths in the Makefile. Previously I was using the clang provided by Debian. Of course I still need to test the result of the build to be sure it works.